db/migrate/20181201000100_create_message_recipients.foodsoft_messages.rb

```diff
@@ -0,0 +1,49 @@
+class CreateMessageRecipients < ActiveRecord::Migration
+  class Message < ActiveRecord::Base
+    has_many :message_recipients
+    has_many :recipients, through: :message_recipients, source: :user
+  end
+
+  def up
+    create_table :message_recipients do |t|
+      t.references :message, index: true, null: false
+      t.references :user, null: false
+      t.integer :email_state, default: 0, null: false
+      t.datetime :read_at
+    end
+
+    add_index :message_recipients, [:user_id, :read_at]
+
+    Message.all.each do |m|
+      recipients = YAML.load(m.recipients_ids).map do |r|
+        {
+          message_id: m.id,
+          user_id: r,
+          email_state: :sent
+        }
+      end
+      MessageRecipient.create! recipients
+    end
+
+    change_table :messages do |t|
+      t.remove :recipients_ids
+      t.remove :email_state
+    end
+  end
+
+  def down
+    change_table :messages do |t|
+      t.text :recipients_ids
+      t.integer :email_state, default: 0, null: false
+    end
+
+    messages = Message.all.includes(:message_recipients).map do |m|
+      recipients = m.message_recipients.map(&:user_id)
+      m.recipients_ids = recipients.to_yaml
+      m.email_state = 1
```

Comment:
maybe good to add a comment what the meaning of `1` is here (in case the enum would migrate later)